.github/workflows/ci.yml

```diff
@@ -0,0 +1,130 @@
+      run: black --check --verbose .
+    - name: Run flake8
+      run: flake8 --exclude docs --statistics
+    - name: Run pydocstyle
+      run: pydocstyle praw
+    - name: Run sphinx
+      run: sphinx-build -W docs/ /tmp/foo
+    strategy:
+      matrix:
+        os: [macOS-latest, ubuntu-latest, windows-latest]
+
+  lint-multi-python:
+    name: Lint Python ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-python@v1
+      with:
+        python-version: 3.x
+    - uses: actions/cache@v1
+      with:
+        key: v0-${{ runner.os }}-pip-lint-${{ hashFiles('setup.py') }}
+        path: ~/.cache/pip
+        restore-keys: |
+          v0-${{ runner.os }}-pip-lint-
+          v0-${{ runner.os }}-pip-
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install .[lint]
+    - name: Run black
+      run: black --check --verbose .
+    - name: Run flake8
+      run: flake8 --exclude docs --statistics
+    - name: Run pydocstyle
+      run: pydocstyle praw
+    - name: Run sphinx
+      run: sphinx-build -W docs/ /tmp/foo
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
+
+  test-multi-os:
+    name: Test ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-python@v1
+      with:
+        python-version: 3.x
+    - uses: actions/cache@v1
+      with:
+        key: v0-${{ runner.os }}-pip-test-${{ hashFiles('setup.py') }}
+        path: ~/.cache/pip
+        restore-keys: |
+          v0-${{ runner.os }}-pip-test-
+          v0-${{ runner.os }}-pip-
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install .[ci,test]
+    - name: Test with pytest
+      run: pytest
+    strategy:
+      matrix:
+        os: [macOS-latest, ubuntu-latest, windows-latest]
+
+  test-multi-python:
+    name: Test Python ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v1
+    - uses: actions/setup-python@v1
+      with:
+        python-version: 3.x
+    - uses: actions/cache@v1
+      with:
+        key: v0-${{ runner.os }}-pip-test-${{ hashFiles('setup.py') }}
+        path: ~/.cache/pip
+        restore-keys: |
+          v0-${{ runner.os }}-pip-test-
+          v0-${{ runner.os }}-pip-
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install .[test]
+        pip install https://github.com/bboe/coveralls-python/archive/github_actions.zip
+    - name: Test with pytest
+      run: coverage run --source praw --module pytest
+    - env:
+        COVERALLS_PARALLEL: true
+        COVERALLS_REPO_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      name: Submit to coveralls
+      run: coveralls
+    strategy:
+      matrix:
+        python-version: [3.5, 3.6, 3.7, 3.8]
+
+name: CI
+on: [pull_request, push]
```

Comment:
Why is everything after this blank?

Reply:
I don't follow. It's the end of the file.

Reply:
i understand, but shouldn't these be at the top of the file?

Reply:
YAML key order doesn't matter. As a result, like all other hash-type definitions, I prefer them to be sorted so that there is little confusion about how things should be ordered (always sort).